python_api/pyiode.cpp

```diff
@@ -124,14 +125,28 @@ NB_MODULE(iode, m)
       m.def("ws_save_var", [](const std::string& filename){ Variables.save(filename); },  nb::arg("filename"), 
             "Save the current variables workspace");
 
+      // IODE object classes
+
+      // Read https://nanobind.readthedocs.io/en/latest/classes.html#classes 
+      // to see how to export C++ classes to Python
+      
+      nb::class_<Scalar>(m, "Scalar")
+            .def_rw("value", &Scalar::val)
+            .def_rw("relax", &Scalar::relax)
+            .def_ro("std", &Scalar::std)
+            .def(nb::init<const std::string&>())
+            .def(nb::init<const IODE_REAL, const IODE_REAL, const IODE_REAL>())
+            .def(nb::self == nb::self)
+            .def("__repr__", &Scalar::to_string);
```

Comment:
@jmpplan 
Above is how to export a C++ class to Python using `nanobind`. 
It is quite easy as long as the C++ class is well written.
Note that I set the `std` attribute of Scalar to be read-only (users should not be able to modify it).
I also overloaded the C++ `==` operator.